### PR TITLE
Modal version

### PR DESCRIPTION
**Pull request by hardikk08**

### CODE REVIEW

File: src/sezzle.js [hunk 1]
```diff
@@ -961,6 +973,33 @@ SezzleJS.prototype.getCSSVersionForMerchant = function (callback) {
...
+    }
+  }.bind(this);
+
+  // Convert document.modalAvailableLanguages into Array
+  let availableLanguages = document.modalAvailableLanguages.split(",").map(function(singleLanguage) {
+    return singleLanguage.trim();
+  });
+  let modalLanguage
+  if(availableLanguages.includes(this.browserLanguage)) modalLanguage = this.browserLanguage;
+  else modalLanguage = availableLanguages[0];
```
Comment (jan-wei) [inline src/sezzle.js]: Not guaranteed to be `en`. I would say just hardcode it has `en`.

File: src/sezzle.js [hunk 2]
```diff
@@ -961,6 +973,33 @@ SezzleJS.prototype.getCSSVersionForMerchant = function (callback) {
...
+
+  let url = `https://d3svog4tlx445w.cloudfront.net/shopify-app/assets/` + document.sezzleDefaultModalVersion + '-'  + modalLanguage + '.html';
```
Comment (jan-wei) [inline src/sezzle.js]: No template literals.

File: src/sezzle.js [hunk 3]
```diff
@@ -961,6 +973,33 @@ SezzleJS.prototype.getCSSVersionForMerchant = function (callback) {
...
+        return console.warn("Can't load the modal because the link provided is not found")
+      }
+    }
+  }.bind(this);
+
+  // Convert document.modalAvailableLanguages into Array
+  let availableLanguages = document.modalAvailableLanguages.split(",").map(function(singleLanguage) {
+    return singleLanguage.trim();
+  });
+  let modalLanguage
```
Comment (jan-wei) [inline src/sezzle.js]: no `let`, use `var`

File: src/sezzle.js [hunk 4]
```diff
@@ -961,6 +973,33 @@ SezzleJS.prototype.getCSSVersionForMerchant = function (callback) {
...
+        callback(html);
+      }
+      else {
+        return console.warn("Can't load the modal because the link provided is not found")
+      }
+    }
+  }.bind(this);
+
+  // Convert document.modalAvailableLanguages into Array
+  let availableLanguages = document.modalAvailableLanguages.split(",").map(function(singleLanguage) {
```
Comment (jan-wei) [inline src/sezzle.js]: no `let`, use `var`

File: src/helper.js [hunk 5]
```diff
@@ -23,8 +23,8 @@ const propsNotInConfigGroup = [
  */
 const widgetLanguageTranslation = function (browserLanguage, numberOfPayments) {
   const translations = {
-    'en': `or ${numberOfPayments} interest-free payments of %%price%% with %%logo%% %%info%%`,
-    'fr': `ou ${numberOfPayments} paiements sans intérêt de %%price%% avec %%logo%% %%info%%`
+    'en': 'or' + numberOfPayments + ' interest-free payments of %%price%% with %%logo%% %%info%%',
```
Comment (jan-wei) [inline src/helper.js]: should have a spacing after `or`

File: src/helper.js [hunk 6]
```diff
@@ -23,8 +23,8 @@ const propsNotInConfigGroup = [
...
+    'fr': 'ou' + numberOfPayments + ' paiements de %%price%% sans intérêts avec %%logo%% %%info%%'
```
Comment (jan-wei) [inline src/helper.js]: spacing here too

File: src/sezzle.js [hunk 7]
```diff
@@ -44,7 +44,7 @@ var SezzleJS = function (options) {
   // Widget Language
   this.browserLanguage = navigator.language || navigator.browserLanguage || 'en';
   this.browserLanguage = this.browserLanguage.toLowerCase().substring(0, 2);
```
Comment (jan-wei) [inline src/sezzle.js]: Reverse `toLowerCase()` and `substring()` (`this.browserLanguage = this.browserLanguage.substring(0, 2).toLowerCase();`)

File: src/sezzle.js [hunk 8]
```diff
@@ -772,33 +772,40 @@ SezzleJS.prototype.renderModal = function () {
     modalNode.className = 'sezzle-checkout-modal-lightbox close-sezzle-modal';
     modalNode.style.display = 'none';
     if (this.altModalHTML) {
-      modalNode.innerHTML = this.altModalHTML;
+      appendModal(this.altModalHTML)
```
Comment (jan-wei) [inline src/sezzle.js]: No client overrides for modal. Remove `this.altModalHTML` altogether. I've discussed this with Marta. The modal should always be fetched from the CDN

Comment (hardikk08) [reply src/sezzle.js]: Oh okay

File: src/sezzle.js [hunk 9]
```diff
@@ -772,33 +772,40 @@ SezzleJS.prototype.renderModal = function () {
...
-  // Event listener to prevent close in modal if click happens within sezzle-checkout-modal
-  let sezzleModal = document.getElementsByClassName('sezzle-modal')[0]
-  // backwards compatability check
-  if (!sezzleModal) sezzleModal = document.getElementsByClassName('sezzle-checkout-modal')[0]
-  sezzleModal.addEventListener('click', function (event) {
-    // stop propagating the event to the parent sezzle-checkout-modal-lightbox to prevent the closure of the modal
-    event.stopPropagation();
-  });
+    // Event listener to prevent close in modal if click happens within sezzle-checkout-modal
+    let sezzleModal = document.getElementsByClassName('sezzle-modal')[0]
```
Comment (jan-wei) [inline src/sezzle.js]: no `let`

File: src/sezzle.js [hunk 10]
```diff
@@ -961,6 +970,33 @@ SezzleJS.prototype.getCSSVersionForMerchant = function (callback) {
...
+
+  // Convert document.modalAvailableLanguages into Array
+  let availableLanguages = document.modalAvailableLanguages.split(",").map(function(singleLanguage) {
+    return singleLanguage.trim();
+  });
+  let modalLanguage
+  if(availableLanguages.includes(this.browserLanguage)) modalLanguage = this.browserLanguage;
+  else modalLanguage = availableLanguages[0];
+
+  let url = `https://d3svog4tlx445w.cloudfront.net/shopify-app/assets/` + document.sezzleDefaultModalVersion + '-'  + modalLanguage + '.html';
```
Comment (jan-wei) [inline src/sezzle.js]: `document.sezzleDefaultModalVersion` comes as something like this: `sezzle-modal1.0.0-{%s%}.html`. You should replace the `{%s%}` with the language. The modal version should be equal to `document.sezzleDefaultModalVersion.replace("{%s%}", modalLanguage)`

Comment (hardikk08) [reply src/sezzle.js]: Ya ya right, was working with local and forgot

File: src/sezzle.js [hunk 11]
```diff
@@ -961,6 +966,33 @@ SezzleJS.prototype.getCSSVersionForMerchant = function (callback) {
...
+        callback(html);
+      }
+      else {
+        return console.warn("Can't load the modal because the link provided is not found")
+      }
+    }
+  }.bind(this);
+
+  // Convert document.modalAvailableLanguages into Array
+  var availableLanguages = document.modalAvailableLanguages.split(",").map(function(singleLanguage) {
```
Comment (jan-wei) [inline src/sezzle.js]: It's always better to check whether the variable is defined before calling methods on them
```javascript
if(document.modalAvailableLanguages) {
      ...
} else {
      ...
}
```

File: src/sezzle.js [hunk 12]
```diff
@@ -961,6 +966,33 @@ SezzleJS.prototype.getCSSVersionForMerchant = function (callback) {
...
+    return singleLanguage.trim();
+  });
+  var modalLanguage
+  if(availableLanguages.includes(this.browserLanguage)) modalLanguage = this.browserLanguage;
+  else modalLanguage = 'en';
+
+  var url = 'https://d3svog4tlx445w.cloudfront.net/shopify-app/assets/' + document.sezzleDefaultModalVersion.replace("{%s%}", modalLanguage) + '-'  + modalLanguage + '.html';
```
Comment (jan-wei) [inline src/sezzle.js]: Perform the same variable checks here

File: src/sezzle.js [hunk 13]
```diff
@@ -961,6 +966,33 @@ SezzleJS.prototype.getCSSVersionForMerchant = function (callback) {
...
+      }
+    }
+  }.bind(this);
+
+  // Convert document.modalAvailableLanguages into Array
+  var availableLanguages = document.modalAvailableLanguages.split(",").map(function(singleLanguage) {
+    return singleLanguage.trim();
+  });
+  var modalLanguage
+  if(availableLanguages.includes(this.browserLanguage)) modalLanguage = this.browserLanguage;
```
Comment (jan-wei) [inline src/sezzle.js]: Don't use `Array.prototype.includes()`, this is only supported in ES6. Do it like this instead
```javascript
if(availableLanguages.indexOf(this.browserLanguage) !== -1) modalLanguage = this.browserLanguage;
```

**Comment by sanbornsen** [comment]
Got a successor (#567) hence closing

**Pull request closed without merging**